mautic/mautic
# Devices report

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR adds new Devices source to the reports.
![obraz](https://user-images.githubusercontent.com/8580942/187607485-5f265f06-6935-4cf5-a2f6-565b5d9a17a1.png)



#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create new Devices report
3. Add all or some devices columns to report
4. Save and see that the data are shown and are correct.
5. Try to filter and group by some devices columns to ensure these features work.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11422"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
